forc-plugins/forc-node/src/ignition/op.rs

```diff
@@ -0,0 +1,106 @@
+use super::cmd::IgnitionCmd;
+use crate::{
+    chain_config::{check_and_update_chain_config, ChainConfig},
+    consts::{
+        MAINNET_BOOTSTRAP_NODE, MAINNET_RELAYER_DA_DEPLOY_HEIGHT,
+        MAINNET_RELAYER_LISTENING_CONTRACT, MAINNET_RELAYER_LOG_PAGE_SIZE, MAINNET_SERVICE_NAME,
+        MAINNET_SYNC_HEADER_BATCH_SIZE,
+    },
+    run_opts::{DbType, RunOpts},
+    util::{ask_user_keypair, ask_user_string, HumanReadableCommand, KeyPair},
+};
+use anyhow::Context;
+use forc_tracing::println_green;
+use std::{
+    net::IpAddr,
+    path::PathBuf,
+    process::{Child, Command},
+};
+/// Configures the node with testnet configuration to connect the node to latest testnet.
+/// Returns `None` if this is a dry_run and no child process created for fuel-core.
+pub(crate) async fn run(cmd: IgnitionCmd, dry_run: bool) -> anyhow::Result<Option<Child>> {
+    check_and_update_chain_config(ChainConfig::Testnet).await?;
+    let keypair = if let (Some(peer_id), Some(secret)) = (
+        &cmd.connection_settings.peer_id,
+        &cmd.connection_settings.secret,
+    ) {
+        KeyPair {
+            peer_id: peer_id.to_string(),
+            secret: secret.to_string(),
+        }
+    } else {
+        ask_user_keypair()?
+    };
+
+    let relayer = if let Some(relayer) = cmd.connection_settings.relayer {
+        relayer
+    } else {
+        ask_user_string("Ethereum RPC (mainnet) Endpoint:")?
+    };
+
+    let opts = IgnitionOpts {
+        keypair,
+        relayer,
+        ip: cmd.connection_settings.ip,
+        port: cmd.connection_settings.port,
+        peering_port: cmd.connection_settings.peering_port,
+        db_path: cmd.db_path,
+    };
+    let run_opts = RunOpts::from(opts);
+    let params = run_opts.generate_params();
+    let mut fuel_core_command = Command::new("fuel-core");
+    fuel_core_command.arg("run");
+    fuel_core_command.args(params.as_slice());
+
+    println_green(&format!(
+        "{}",
+        HumanReadableCommand::from(&fuel_core_command)
+    ));
+
+    if dry_run {
+        Ok(None)
+    } else {
+        // Spawn the process with proper error handling
+        let handle = fuel_core_command
+            .spawn()
+            .with_context(|| "Failed to spawn fuel-core process:".to_string())?;
+        Ok(Some(handle))
+    }
```

Comment:
nit: can remove nesting (if-else):

```suggestion
    if dry_run {
        return Ok(None);
    }
    // Spawn the process with proper error handling
    let handle = fuel_core_command
        .spawn()
        .with_context(|| "Failed to spawn fuel-core process:".to_string())?;
    Ok(Some(handle))
```